cli/src/torii/routing.rs

```diff
@@ -119,15 +119,37 @@ pub(crate) async fn handle_queries(
     wsv: Arc<WorldStateView>,
     query_judge: QueryJudgeArc,
     pagination: Pagination,
+    sorting: Sorting,
     request: VerifiedQueryRequest,
 ) -> Result<Scale<VersionedPaginatedQueryResult>> {
     let (valid_request, filter) = request.validate(&wsv, query_judge.as_ref())?;
     let original_result = valid_request.execute(&wsv)?;
     let result = filter.filter(original_result);
-    let (total, result) = if let Value::Vec(value) = result {
+    let (total, result) = if let Value::Vec(mut vec) = result {
+        if let Some(ref key) = sorting.sort_by_metadata_key {
+            let f = |value1: &Value| {
+                if let Value::U128(num) = value1 {
+                    *num
+                } else {
+                    0
+                }
+            };
+
+            vec.sort_by_key(|value0| match value0 {
+                Value::Identifiable(IdentifiableBox::Asset(asset)) => match asset.value() {
+                    AssetValue::Store(store) => store.get(key).map_or(0, f),
+                    _ => 0,
+                },
+                Value::Identifiable(v) => TryInto::<Box<&dyn HasMetadata>>::try_into(v)
```

Comment:
does it have to be boxed? Maybe you could have written: `TryInto::<&dyn HasMetadata>::try_into(v)`

Reply:
also there is `TryAsRef` in the `data_model` that would make sense to make use of here

Reply:
To be honest, I don't have any idea how to implement `TryFrom` for raw `&dyn HasMetadata`. [See](https://github.com/hyperledger/iroha/pull/2575/files#diff-6af9c278c26aa09da5007e70f12f7210072f252c41cab1e09aef22932ac8856dR401)

Reply:
Unfortunately, `TryAsRef` implemented only for `AssetValue` not for `IdentifiableBox`.

Reply:
> To be honest, I don't have any idea how to implement TryFrom for raw &dyn HasMetadata. See

[see](https://github.com/hyperledger/iroha/pull/2575/files#r941015465)

> Unfortunately, TryAsRef implemented only for AssetValue not for IdentifiableBox.

I meant to say that it might be possible to implement that trait for `IdentifiableBox`

Reply:
> [see](https://github.com/hyperledger/iroha/pull/2575/files#r941015465)
damn, it works...

> I meant to say that it might be possible to implement that trait for IdentifiableBox. I don't know why do we have that trait when TryFrom would do as fine when implemented for references
Ah, I got you, absolutely agree.